### PR TITLE
chore(android): update cmake, checkstyle

### PR DESCRIPTION
**Pull request by m1ga**
* update **cmake to 3.22.1** (as it looks like that is available on all systems), Short note: https://www.kitware.com/cmake-3-22-1-available-for-download/ - Full notes: https://cmake.org/cmake/help/latest/release/3.22.html
* update **checkstyle to 10.11.0**: https://checkstyle.org/releasenotes.html (next version will currently fail)
* updating all Java files to full-fill checkstyles rules (no code changes, just linting!)
* update allowed venderVersions in android/package.json
 
**Note:**
* all Java file changes are just automatic linting in Android Studio! While updating checkstyle from 8 to 10 some rules changed so I had to manually open each file and run Android Studio lint (ctrl + shift + alt + L).

**Tests:**
* SDK builds fine
* java module builds fine
* kotlin module builds fine
* app with existing modules builds find and works

Tested on Fedora Linux

### CODE REVIEW

**Comment by m1ga** [comment]
Interesting. The build script returns:
```
[CXX1300] CMake '3.27.7' was not found in SDK, PATH, or by cmake.dir property.
[CXX1301] - CMake '3.28.1' found in PATH did not satisfy requested version.
[CXX1301] - CMake '3.10.2' found in SDK did not satisfy requested version.
[CXX1301] - CMake '3.18.1' found in SDK did not satisfy requested version.
[CXX1301] - CMake '3.22.1' found in SDK did not satisfy requested version.

[CXX1300] CMake '3.27.7' was not found in SDK, PATH, or by cmake.dir property.
[CXX1301] - CMake '3.28.1' found in PATH did not satisfy requested version.
[CXX1301] - CMake '3.10.2' found in SDK did not satisfy requested version.
[CXX1301] - CMake '3.18.1' found in SDK did not satisfy requested version.
[CXX1301] - CMake '3.22.1' found in SDK did not satisfy requested version.
```

I'll check that

**Comment by cb1kenobi** [comment]
Is this PR ready for review? Is there any specific steps to test?

**Comment by m1ga** [comment]
yes, you can review/test it. The tests I did are the one in the first post: build sdk, build a module (java and kotlin), build an app, build an app with 12.2.1 modules.

If you build with this PR it should download cmake and checkstyle during the first build if they are not installed already. Then you shouldn't see any lint error during the build (that's why there are the java file changes). checkstyle will run during the build process and throw an error if the java files aren't in the correct style.

So to summarize: if you can build the SDK and modules + app: everything works fine after this PR :smile:

**Review by cb1kenobi — APPROVED** [review]
Tested and looks good! **APPROVED**
